types/bluebird/bluebird-tests.ts

```diff
@@ -373,6 +373,30 @@ fooProm = fooProm.catch(CustomError, reason => {
 	fooProm = fooProm.catch(CustomError1, CustomError2, CustomError3, error => {});
 	fooProm = fooProm.catch(CustomError1, CustomError2, CustomError3, CustomError4, error => {});
 	fooProm = fooProm.catch(CustomError1, CustomError2, CustomError3, CustomError4, CustomError5, error => {});
+
+	const booPredicate1 = (error: CustomError1) => true;
+	const booPredicate2 = (error: [number]) => true;
+	const booPredicate3 = (error: string) => true;
+	const booPredicate4 = (error: Object) => true;
+	const booPredicate5 = (error: any) => true;
+
+	fooProm = fooProm.catch(booPredicate1, error => {});
+	fooProm = fooProm.catch(booPredicate1, booPredicate2, error => {});
+	fooProm = fooProm.catch(booPredicate1, booPredicate2, booPredicate3, error => {});
+	fooProm = fooProm.catch(booPredicate1, booPredicate2, booPredicate3, booPredicate4, error => {});
+	fooProm = fooProm.catch(booPredicate1, booPredicate2, booPredicate3, booPredicate4, booPredicate5, error => {});
+
+	const booObject1 = new CustomError1();
+	const booObject2 = [400, 500];
+	const booObject3 = ["Error1", "Error2"];
+	const booObject4 = {code: 400};
+	const booObject5: any = null;
+
+	fooProm = fooProm.catch(booObject1, error => {});
+	fooProm = fooProm.catch(booObject1, booObject2, error => {});
+	fooProm = fooProm.catch(booObject1, booObject2, booObject3, error => {});
+	fooProm = fooProm.catch(booObject1, booObject2, booObject3, booObject4, error => {});
+	fooProm = fooProm.catch(booObject1, booObject2, booObject3, booObject4, booObject5, error => {});
```

Comment:
@lhecker yes it is certainly better. The problem is that it is still unsound.

```js
class CustomError1 extends Error {
  static foo : number = 1
  constructor (message) {
    super(message)
    this.foo : string = 'bar'
  }
}

fooProm.catch(CustomError1, booObject2, booPredicate3, (error) => {
  error.foo // expecting a string, TS thinks it's a number
})
```

I'm not sure which is worse for the developer. Having TS incorrectly refine the `error` or not having TS refine it at all (e.g. `any`).